Cognite.Simulator.Utils/SimulationScheduler.cs

```diff
@@ -83,91 +103,158 @@ public class SimulationSchedulerBase<U, V>
+                                    Config = config
 
-                            configState.LastRun = nextRun.ToUnixTimeMilliseconds(); // store state
-                            var runEvent = CreateRunEvent(configState, configObj); // create CDF event body
-                            eventsToCreate.Add(runEvent);
-                            break;
+                                };
+                                _logger.LogDebug("Created new job for schedule: {0} with id {1}", config.Schedule.Repeat, config.ExternalId);
+                                scheduledJobs.Add(config.CalculationName, newJob);
+                            }
+                            catch (Exception e)
+                            {
+                               _logger.LogError($"Exception while scheduling job for Calculation : {job.CalculationName} Error: {e.Message}. Skipping.");
+                            }
                         }
                     }
+
+                    // Schedule new jobs
+                    List<Task> tasks = new List<Task>();
+                    foreach (var kvp in scheduledJobs)
+                    {
+                        if (kvp.Value.Scheduled)
+                        {
+                            continue;
+                        }
+                        Console.WriteLine($"Scheduling job for: {kvp.Key}");
+                        tasks.Add(RunJob(kvp.Value, _config.GetConnectorName(), token));
+                        kvp.Value.Scheduled = true;
+                    }
+                    if (tasks.Count != 0)
+                    {
+                        Task.WhenAll(tasks);
+                    } 
+                    // Wait for 10 seconds before checking again
+                    await Task.Delay(interval);
+                }
+            }).ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Runs a scheduled job.
+        /// </summary>
+        /// <param name="job">The scheduled job to run.</param>
+        /// <param name="connectorName"></param>
+        /// <param name="mainToken">The cancellation token.</param>
+        public async Task RunJob(ScheduledJob<U,V> job, string connectorName, CancellationToken mainToken)
+        {
+            if (job == null)
+            {
+                _logger.LogError($"Scheduled Job is null. Exiting.");
+                return;
+            }
+            while (!mainToken.IsCancellationRequested || !job.TokenSource.Token.IsCancellationRequested)
+            {
+                var nextOccurrence = job.Schedule.GetNextOccurrence(DateTime.Now);
+                var delay = nextOccurrence - DateTime.Now;
+                // Retrieve the last run time saved in the calculation state, or use the start date
+                // if no run was saved in the state
+                if (job.ConfigState.LastRun.HasValue ) {
+                    var lastRun = CogniteTime.FromUnixTimeMilliseconds(job.ConfigState.LastRun.Value) ;
```

Comment:
this line is not used?

Reply:
If we don't use `lastRun` anymore so I think we should try to remove the `ConfigurationStateBase` now. (the local db)